chore: Document workaround for Raspberry Pi 3 dunfel to kirkstone update (#1832)
mendersoftware/mender-docs: 301.Troubleshoot/03.Mender-Client/docs.md

    * `rootfs-image.checksum: b9147deb5` - same as the `v2 mender-binary-delta`

### How to check this on the device/server/artifact?

* artifact - `mender-artifact read <mender-artifact.mender>`
* device - Run the command on the device `mender show-provides`
* server UI - `Releases -> Select Release -> Expand the artifact info by clicking it -> Expand Provides and Depends`

<!--AUTOVERSION: "from `%` to `%` on"/ignore -->
### Deploying an update from `dunfell` to `kirkstone` on Raspberry Pi blocks forever with `Waiting for root device /dev/mmcblk0p3...`

Raspberry Pi boards have a set of boot firmware files (including Device Tree Blob files) that are
located on the vfat boot part, and these files are not updated when you perform an update of the
root filesystem.

> Occasionally there will be changes to the Raspberry Pi software stack that requires that these files
> are update. Typically there is something in the Linux kernel that requires something that is in the
> boot firmware. This means that to update the Linux kernel you must also update the boot firmware
> files.

Reviewer: ```suggestion
Occasionally, changes to the Raspberry Pi software stack require that these files are
updated. Typically there are dependencies in the Linux kernel on something in the boot
firmware. Therefore, to update the Linux kernel, you must also update the boot firmware
files.
```

<!--AUTOVERSION: "between `%` and `%` yocto"/ignore -->
This is what has happened between `dunfell` and `kirkstone` yocto releases for Raspberry Pi 3B
(other boards might have been affected too).

> Mender provides a recipe to include an state script in the Artifact that will copy the firmware

Reviewer: ```suggestion
Mender provides a recipe to include a state script in the Artifact that will copy the firmware
```

files into the boot during the Artifact install. Enable it by adding the following to e.g
`local.conf`:

```
INHERIT += "rpi-update-firmware"
```

> The script can be overridden to customize what
> files to update, e.g only DTB files. See [mendersoftware/meta-mender@5abe2e7](https://github.com/mendersoftware/meta-mender/commit/5abe2e7b1f85b3482cb53b747ae9ced3f20a3e73) for more details.

Reviewer: ```suggestion
You can override the script customizing what files to update, e.g only DTB files.
See [mendersoftware/meta-mender@5abe2e7](https://github.com/mendersoftware/meta-mender/commit/5abe2e7b1f85b3482cb53b747ae9ced3f20a3e73) for more details.
```

> !!! Updating the boot firmware files can not be done atomically and the files are not roll-backed
> !!! even though an rootfs rollback is performed. Conclusion, this is an risky operating which could
> !!! brick your device. We recommend to use this workaround once, when upgrading between yocto
> !!! releases, and not for every update henceforth.

Reviewer: ```suggestion
!!! Updating the boot firmware files can not be done atomically and the files are not rollbacked
!!! even though a rootfs rollback is performed. In conclusion, this is a risky operating which could
!!! brick your device. We recommend using this workaround once when upgrading between yocto
!!! releases, and not for every update henceforth.
```

    * `rootfs-image.checksum: b9147deb5` - same as the `v2 mender-binary-delta`

### How to check this on the device/server/artifact?

* artifact - `mender-artifact read <mender-artifact.mender>`
* device - Run the command on the device `mender show-provides`
* server UI - `Releases -> Select Release -> Expand the artifact info by clicking it -> Expand Provides and Depends`

<!--AUTOVERSION: "from `%` to `%` on"/ignore -->
### Deploying an update from `dunfell` to `kirkstone` on Raspberry Pi blocks forever with `Waiting for root device /dev/mmcblk0p3...`

Raspberry Pi boards have a set of boot firmware files (including Device Tree Blob files) that are

> located on the vfat boot part, and these files are not updated when you perform an update of the

Reviewer 2: "boot partition" please, "boot part" is jargon.

> located on the vfat boot part, and these files are not updated when you perform an update of the

Reviewer 3: Q: Why vfat explicitly?

Author: I copied it from the commit message. I'll remove it.

> located on the vfat boot part, and these files are not updated when you perform an update of the

Reviewer 2: "boot partition" please, "boot part" is jargon

> root filesystem.

Reviewer 3: S: Add something like: `where the kernel lives`.

Occasionally, changes to the Raspberry Pi software stack require that these files are updated.

> Typically there are dependencies in the Linux kernel on something in the boot firmware. Therefore,

Reviewer 2: "on something" sounds quite handwavy. Maybe better "Between the Linux kernel and the boot firmware, version interdependencies exist which need to be taken care of."

to update the Linux kernel, you must also update the boot firmware files.

<!--AUTOVERSION: "between `%` and `%` yocto"/ignore -->
This is what has happened between `dunfell` and `kirkstone` yocto releases for Raspberry Pi 3B
(other boards might have been affected too).

Mender provides a recipe to include a state script in the Artifact that will copy the firmware
files into the boot during the Artifact install. Enable it by adding the following to e.g
`local.conf`:

```
INHERIT += "rpi-update-firmware"
```

You can override the script customizing what files to update, e.g only DTB files. See
[mendersoftware/meta-mender@5abe2e7](https://github.com/mendersoftware/meta-mender/commit/5abe2e7b1f85b3482cb53b747ae9ced3f20a3e73)
for more details.

> !! Updating the boot firmware files can not be done atomically and the files are not rollbacked

Reviewer 3: S: `rolled back`

> !! Updating the boot firmware files can not be done atomically and the files are not rollbacked

Reviewer 2: "are not rollbacked" -> "cannot be rolled back even if a rootfs rollback might be performed"

> !! even though a rootfs rollback is performed. In conclusion, this is a risky operating which could

Reviewer 3: `s/operating/operation`

> !! brick your device. We recommend using this workaround once when upgrading between Yocto

Reviewer 2: "only once"
